hack/load-image.sh

```diff
@@ -51,6 +52,14 @@ for file in config/manager/manager.yaml examples/operator/operator.yaml ; do
     "$file"
 done
 
+# Update the operator's image.
+for file in config/manager/manager.yaml examples/operator/operator.yaml ; do
+  echo "setting \"image: ${IMAGE}:${VERSION}\" for $file"
+  run::sed \
+    "-es|image: ${IMAGE}:${OLD_VERSION}|image: ${IMAGE}:${VERSION}|" \
+    "$file"
+done
+
```

Comment:
FWIW in `projectcontour/contour`, when we build an image for E2E tests, we tag it as `main` so that we don't have to change the manifest's image tags for e2e test purposes. If I'm following this correctly, we have to change the manifests here because the built image is tagged with the git commit sha.

I'm not opposed to doing it this way for the operator, especially since I'm not sure what other knock-on effects tagging as `main` would have, just wanted to provide the info on how `projectcontour/contour` does it in case it's useful.

See https://github.com/projectcontour/contour/blob/main/_integration/testsuite/install-contour-working.sh#L32-L36 and https://github.com/projectcontour/contour/blob/main/_integration/testsuite/install-contour-working.sh#L53-L65 for ref.

Reply:
> If I'm following this correctly, we have to change the manifests here because the built image is tagged with the git commit sha.

@skriss that is correct. This allows users to run the operator in a cluster using the built image tag (for testing purposes).